### ISSUE TITLE
warning: method redefined; discarding old use_eventmachine=

### ISSUE THREAD

**Issue by fguillen** [rollbar/rollbar-gem#544]
Any idea how to solve these warnings?

```
/Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rollbar-2.13.3/lib/rollbar/configuration.rb:194: warning: method redefined; discarding old use_eventmachine=
/Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rollbar-2.13.3/lib/rollbar/configuration.rb:209: warning: method redefined; discarding old before_process=
/Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rollbar-2.13.3/lib/rollbar/configuration.rb:213: warning: method redefined; discarding old transform=
/Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rollbar-2.13.3/lib/rollbar/configuration.rb:232: warning: method redefined; discarding old logger
/Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rollbar-2.13.3/lib/rollbar/notifier.rb:1: warning: loading in progress, circular require considered harmful - /Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rollbar-2.13.3/lib/rollbar.rb
```

**Comment by jondeandres** [comment]
@fguillen I don't have any of this right now. Are you monkey patching our gem somewhere?

**Comment by fguillen** [comment]
@jondeandres no I am not.

More info:

```
$ bundle list
Gems included by the bundle:
  * ab_testi_client (0.1.0 2e11b00)
  * activemodel (5.0.0.1)
  * activerecord (5.0.0.1)
  * activesupport (5.0.0.1)
  * airbrussh (1.1.1)
  * arel (7.1.4)
  * backports (3.6.8)
  * bundler (1.12.5)
  * capistrano (3.6.1)
  * capistrano-bundler (1.2.0)
  * capistrano-harrow (0.5.3)
  * capistrano-rbenv (2.0.4 e056efc)
  * concurrent-ruby (1.0.2)
  * curb (0.9.3)
  * daemons (1.2.4)
  * dalia_api_campaign_platform (0.1.3 1c8a99e)
  * dalia_api_panel_hub (0.0.7 bba2b9a)
  * eventmachine (1.2.0.1)
  * git-version-bump (0.15.1)
  * httparty (0.14.0)
  * i18n (0.7.0)
  * json (2.0.2)
  * macaddr (1.7.1)
  * metaclass (0.0.4)
  * mini_graphite (0.1.1 353ab48)
  * minitest (5.9.1)
  * mocha (1.2.1)
  * multi_json (1.12.1)
  * multi_xml (0.5.5)
  * mysql2 (0.3.21)
  * net-scp (1.2.1)
  * net-ssh (3.2.0)
  * rack (1.6.4)
  * rack-contrib (1.4.0)
  * rack-protection (1.5.3)
  * rack-test (0.6.3)
  * rake (11.3.0)
  * recursive-open-struct (1.0.1)
  * redcarpet (3.3.4)
  * rollbar (2.13.3)
  * sinatra (1.4.7)
  * sinatra-activerecord (2.0.11)
  * sinatra-contrib (1.4.7)
  * sshkit (1.11.4)
  * systemu (2.6.5)
  * thin (1.7.0)
  * thread_safe (0.3.5)
  * tilt (2.0.5)
  * tzinfo (1.2.2)
  * uuid (2.3.8)
```

_For my reference this is the PP project_

**Comment by jondeandres** [comment]
Hi @fguillen, and you stubbing at some point something related to Rollbar on those tests? It seems it may be something with that?

**Comment by fguillen** [comment]
Hi @jondeandres I managed to reproduce the issue in the smallest version of my project possible: 

- https://github.com/fguillen/RollbarWarningTest

Try to

    bundle install
    rake

When I do this I see: 

```
$ rake
/Users/fguillen/.rbenv/versions/2.3.1/bin/ruby -w -I"lib:." -I"/Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rake-12.0.0/lib" "/Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rake-12.0.0/lib/rake/rake_test_loader.rb" "test/utils_test.rb" 
/Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rollbar-2.13.3/lib/rollbar/configuration.rb:194: warning: method redefined; discarding old use_eventmachine=
/Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rollbar-2.13.3/lib/rollbar/configuration.rb:209: warning: method redefined; discarding old before_process=
/Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rollbar-2.13.3/lib/rollbar/configuration.rb:213: warning: method redefined; discarding old transform=
/Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rollbar-2.13.3/lib/rollbar/configuration.rb:232: warning: method redefined; discarding old logger
/Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rollbar-2.13.3/lib/rollbar/notifier.rb:1: warning: loading in progress, circular require considered harmful - /Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rollbar-2.13.3/lib/rollbar.rb
	from /Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rake-12.0.0/lib/rake/rake_test_loader.rb:4:in  `<main>'
	from /Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rake-12.0.0/lib/rake/rake_test_loader.rb:4:in  `select'
	from /Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rake-12.0.0/lib/rake/rake_test_loader.rb:15:in  `block in <main>'
	from /Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rake-12.0.0/lib/rake/rake_test_loader.rb:15:in  `require'
	from /Users/fguillen/Development/Temp/RollbarWarningTest/test/utils_test.rb:1:in  `<top (required)>'
	from /Users/fguillen/Development/Temp/RollbarWarningTest/test/utils_test.rb:1:in  `require_relative'
	from /Users/fguillen/Development/Temp/RollbarWarningTest/test/test_helper.rb:5:in  `<top (required)>'
	from /Users/fguillen/Development/Temp/RollbarWarningTest/test/test_helper.rb:5:in  `require'
	from /Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rollbar-2.13.3/lib/rollbar.rb:19:in  `<top (required)>'
	from /Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rollbar-2.13.3/lib/rollbar.rb:19:in  `require'
	from /Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rollbar-2.13.3/lib/rollbar/notifier.rb:1:in  `<top (required)>'
	from /Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rollbar-2.13.3/lib/rollbar/notifier.rb:1:in  `require'
/Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rollbar-2.13.3/lib/rollbar.rb:153: warning: method redefined; discarding old with_config
/Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/2.3.0/forwardable.rb:187: warning: previous definition of with_config was here
/Users/fguillen/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rollbar-2.13.3/lib/rollbar.rb:40: warning: instance variable @root_notifier not initialized
Run options: --seed 53334

# Running:



Finished in 0.001080s, 0.0000 runs/s, 0.0000 assertions/s.

0 runs, 0 assertions, 0 failures, 0 errors, 0 skips
```

**Comment by jondeandres** [comment]
awesome @fguillen! we'll take a look soon

**Comment by robin850** [comment]
Hi there !

Actually, reproducing the problem is as simple as doing `ruby -w -e 'require "rollbar"'`. A part of the warning has been fixed in https://github.com/rollbar/rollbar-gem/commit/482ff25d318c4e3287360cf2dc8dedc24d89190b but the problem is still visible in @fguillen's project as its `Gemfile.lock` files points to 2.13.3 but this fix is part of 2.14.0.

I don't know how you want to fix that but it looks like the problem is still in `rollbar/notifier.rb`.

Have a nice day !

**Comment by jondeandres** [comment]
Hey @robin850, thanks for writing here. Which warning you comment is still in rollbar/notifier.rb? I've checked this and I think we still need to remove a warning related to a not initialized `@root_notifier` variable.

We suggest to upgrade always to the latest version, since we rarely write patches for previous versions.

BTW, this PR should remove that last warning on `@root_notifier`, https://github.com/rollbar/rollbar-gem/pull/562.

**Comment by robin850** [comment]
@jondeandres : Yeah, actually I'm not using this gem, just commenting since Redcarpet had a similar issue and it was reported by @fguillen as well. :p Actually I forgot to check with 2.14.0 before commenting but you're totally right, the `@root_notifier` warning is the only one remaining. 👏

**Comment by rivkahstandig3636** [comment]
Hi there, I’m closing out all issues opened before 2018 that haven’t had any activity on them since the start of this year. If this is still an issue for you, please comment here and we can reopen this. Thanks!

**Comment by fredngo** [comment]
Hey! Still seeing this:

```
/Users/fred/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rollbar-2.16.4/lib/rollbar/plugins/rails/railtie_mixin.rb:1: warning: loading in progress, circular require considered harmful - /Users/fred/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rollbar-2.16.4/lib/rollbar.rb
```

**Comment by ArturMoczulski** [comment]
@fredngo okay, I'm on it

**Comment by ArturMoczulski** [comment]
After updating rollbar-gem dependency to the latest release `v2.16.4`, I don't see this error anymore in https://github.com/fguillen/RollbarWarningTest

@fredngo can you explain in more details how you are reproducing the problem?

**Comment by fredngo** [comment]
@ArturMoczulski Hi Artur, I am actually seeing this message when writing a Rails Engine inside a RubyGem, and then running tests with `rake`.

This is a fairly complex gem for a client so I can't just post the code, but the basic gist is this:

In `my_engine.gemspec` I have:

```
s.add_dependency 'rollbar', '~> 2.16'
```

and then inside `lib/my_engine/engine.rb` I have:

```
require 'rollbar'
```

I also have a Roller configuration file inside `config/initializers/rollbar.rb`:

```
Rollbar.configure do |config|
  ... bunch of stuff ...
end
```

and finally I have a base ActiveJob class that includes `Roller::ActiveJob` in `app/jobs/my_engine/application_job.rb`

```
class MyEngine::ApplicationJob < ActiveJob::Base
  include Rollbar::ActiveJob
end
```

Hope this helps?

**Comment by ArturMoczulski** [comment]
@fredngo thx! Back to solving this now

**Comment by ArturMoczulski** [comment]
This is fixed now in `master` and will be released in `v2.18.0`
